simulator/src/models/power-plant.ts

```diff
@@ -0,0 +1,564 @@
+                time: this._time,
+                status: this._status,
+
+                productionLevel: this._productionLevel,
+                productionCapacity: this._productionCapacity,
+                productionVariant: this._productionVariant,
+                productionRatio: this.productionRatio,
+
+                batteryValue: this._battery.value,
+                batteryCapacity: this._battery.capacity,
+
+                totalProduction: totalProduction,
+                unit: this._unit,
+            };
+    }
+    
+
+    /**
+     * Gets the id of the power plant.
+     * @returns {uuid.v4} the power plants id.
+     */
+    get id(): uuid.v4 {
+        return this._id;
+    }
+
+
+    /**
+     * Gets the time it takes for the power plant to start in milliseconds.
+     * @returns {uuid.v4} the time it takes for the power plant to start.
+     */
+    get startDelay(): number {
+        return this._startDelay;
+    }
+
+
+    /**
+     * Gets the time it takes for the power plant to stop in milliseconds.
+     * @returns {uuid.v4} the time it takes for the power plant to stop.
+     */
+    get stopDelay(): number {
+        return this._stopDelay;
+    }
+
+
+    /**
+     * Gets the power plants status.
+     * @returns {Status} the power plants status.
+     */
+    get status(): Status {
+        return this._status;
+    }
+
+
+    /**
+     * Gets the productionLevel variable value.
+     * @returns {number} the current productionLevel value.
+     */
+    get productionLevel(): number {
+        return this._productionLevel;
+    }
+
+
+    /**
+     * Gets the productionCapacity variable value.
+     * @returns {number} the current productionCapacity value.
+     */
+    get productionCapacity(): number {
+        return this._productionCapacity;
+    }
+
+
+    /**
+     * Gets the productionVariant variable value.
+     * @returns {number} the current productionVariant value.
+     */
+    get productionVariant(): number {
+        return this._productionVariant;
+    }
+
+
+    /**
+     * Gets the productionRatio variable value.
+     * @returns {number} the current productionRatio value.
+     */
+    get productionRatio(): number {
+        return this._productionRatio;
+    }
+
+
+    /**
+     * Gets the owner of the power plant.
+     * @returns {uuid.v4} the owner of the power plant.
+     */
+    get owner(): uuid.v4 {
+        return this._owner;
+    }
+
+
+    /**
+     * Gets the batteries owner;
+     * @returns {uuid.v4} the owner of the battery.
+     */
+    get batteryOwner(): uuid.v4 {
+        return this._battery.owner;
+    }
+
+
+    /**
+     * Gets the batteries current value.
+     * @returns {number} the batteries current value.
+     */
+    get batteryValue(): number {
+        return this._battery.value;
+    }
+
+
+    /**
+     * Gets the batteries capacity.
+     * @returns {number}
+     */
+    get batteryCapacity(): number {
+        return this._battery.capacity;
+    }
+
```

Comment:
Not useful you can just do powerplant.battery.owner/value/capacity, if these are commonly used consider changing these to normal method as they are not actual fields of this class.

Reply:
Also can you add grant permissions and delete table ... cascade to migration.sh so it is easier to change the database.

Reply:
Sure I can add that to migration.sh. But what is migration.sh used for? Isn't it easier to use `docker-compose down -v` to delete all project instances and volumes?